bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/db/JdbcDerbyDAO.java

```diff
@@ -103,15 +106,15 @@ public void initAfterFirstDbConnection() {
      **************/
     @Override
     public @Nullable Integer doPingDB() {
-        return Yank.queryScalar(sqlPingDB, (Class<@Nullable Integer>) Integer.class, null);
+        return Yank.queryScalar(sqlPingDB, Integer.class, null);
     }
 
     @Override
     public boolean doIfTableExists(ItemsVO vo) {
         String sql = StringUtilsExt.replaceArrayMerge(sqlIfTableExists, new String[] { "#searchTable#" },
                 new String[] { vo.getItemsManageTable().toUpperCase() });
         logger.debug("JDBC::doIfTableExists sql={}", sql);
-        return Yank.queryScalar(sql, (Class<@Nullable String>) String.class, null) != null;
+        return Yank.queryScalar(sql, String.class, null) != null;
```

Comment:
This reintroduces this warning:
```
JdbcDerbyDAO.java:[117,16] Redundant null check: comparing '@NonNull String' against null
```

which was fixed by #13429. Maybe there is a more elegant way of solving it than type-cast?

Reply:
I will have a look or change it back to prevent the warning. Me and my auto save actions thought it was probably a good idea to clean this up as an unneccesary cast. :wink:

Reply:
Thanks. See also https://github.com/openhab/openhab-addons/pull/13429#discussion_r979376665.

Reply:
WDYT about my new commit?

Reply:
It seems safer and at the same time eliminates a potential null type safety warning depending on compiler options, so your version wins. :-) There are four other casts like this (search for "(Class<@Nullable"), could you be persuaded to refactor them too? This way I would stay competent to merge the PR. ;-)